Cannot find module 'web3-types/src/eth_types' or its corresponding type declarations.

I'm getting the titular error from this line:

https://github.com/web3/web3.js/blob/8e3e67628ebc90ffe6b86ec054874cb6bb7bac31/packages/web3-rpc-methods/src/eth_rpc_methods.ts#L31C44-L31C58

Everything works okay if I change that line to
```typescript
import { Eip712TypedData } from 'web3-types';
```

#### Expected behavior
`npx tsc` runs okay

#### Actual behavior
build fails

#### Steps to reproduce the behavior
Source code that crashes during typescript build
```typescript
import Web3 from 'web3';
const web3 = new Web3(new Web3.providers.HttpProvider("https://polygon-mainnet.infura.io/v3/" + process.env.INFURA_API_KEY))
```

`npx tsc`

#### Logs
```bash
$ npx tsc
node_modules/web3-rpc-methods/lib/types/eth_rpc_methods.d.ts:3:33 - error TS2307: Cannot find module 'web3-types/src/eth_types' or its corresponding type declarations.

3 import { Eip712TypedData } from 'web3-types/src/eth_types';
```

#### Environment
node: 20.9.0

package.json dependencies:
```javascript
{
  "engines": {
    "node": ">= 20.9.0"
  },
  "devDependencies": {
    "@types/jest": "^29.5.8",
    "@types/node": "^20.9",
    "@typescript-eslint/eslint-plugin": "^6.10",
    "@typescript-eslint/parser": "^6.10",
    "eslint": "^8.53",
    "eslint-config-prettier": "~9.0",
    "eslint-plugin-jest": "^27.6",
    "jest": "^29.7.0",
    "prettier": "~3.0",
    "rimraf": "~5.0",
    "ts-api-utils": "~1.0",
    "ts-jest": "~29.1",
    "typescript": "^5.2.2"
  },
  "dependencies": {
    "tslib": "~2.6",
    "web3": "^4.2.2",
    "web3-types": "^1.3.1"
  }
}
```

and tsconfig.json
```javascript
{
  "compilerOptions": {
    "target": "es2022",
    "module": "node16",
    "lib": ["ES2022"],
    "moduleResolution": "node16",
    "rootDir": ".",
    "outDir": "build",
    "allowSyntheticDefaultImports": true,
    "importHelpers": true,
    "alwaysStrict": true,
    "sourceMap": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "noImplicitReturns": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitAny": false,
    "noImplicitThis": false,
    "strictNullChecks": false
  },
  "include": ["src/**/*", "__tests__/**/*"]
}

```

--- a/packages/web3-rpc-methods/CHANGELOG.md
+++ b/packages/web3-rpc-methods/CHANGELOG.md
@@ -127,3 +127,7 @@ Documentation:
 -   Dependencies updated
 
 ## [Unreleased]
+
+### Fixed
+
+-   Fix web3-types import #6590 (#6589)

--- a/packages/web3-rpc-methods/src/eth_rpc_methods.ts
+++ b/packages/web3-rpc-methods/src/eth_rpc_methods.ts
@@ -28,7 +28,7 @@ import {
 	Uint256,
 	Web3EthExecutionAPI,
 } from 'web3-types';
-import { Eip712TypedData } from 'web3-types/src/eth_types';
+import { Eip712TypedData } from 'web3-types';
 import { validator } from 'web3-validator';
 
 export async function getProtocolVersion(requestManager: Web3RequestManager) {

--- a/packages/web3-rpc-methods/src/eth_rpc_methods.ts
+++ b/packages/web3-rpc-methods/src/eth_rpc_methods.ts
@@ -27,7 +27,7 @@ import {
 	Uint,
 	Uint256,
 	Web3EthExecutionAPI,
-	Eip712TypedData
+	Eip712TypedData,
 } from 'web3-types';
 import { validator } from 'web3-validator';
 

--- a/packages/web3-rpc-methods/src/eth_rpc_methods.ts
+++ b/packages/web3-rpc-methods/src/eth_rpc_methods.ts
@@ -27,8 +27,8 @@ import {
 	Uint,
 	Uint256,
 	Web3EthExecutionAPI,
+	Eip712TypedData
 } from 'web3-types';
-import { Eip712TypedData } from 'web3-types';
 import { validator } from 'web3-validator';
 
 export async function getProtocolVersion(requestManager: Web3RequestManager) {

